engine/source/public/core/fileio.hpp

```diff
@@ -0,0 +1,47 @@
+﻿#pragma once
+#include <map>
+#include <string_view>
+#include <vector>
+
+namespace lumina
+{
+    class Engine;
+
+    class FileIO
+    {
+    public:
+        FileIO();
+        ~FileIO() = default;
+
+        FileIO(const FileIO&)            = delete;
+        FileIO(FileIO&&)                 = delete;
+        FileIO& operator=(const FileIO&) = delete;
+        FileIO& operator=(FileIO&&)      = delete;
+
+        enum class Directory
+        {
+            Assets,
+            Config,
+            Log
+        };
+
+        //Text file read/write
+        std::string ReadTextFile(Directory directory, const std::string& filePath);
+        bool WriteTextFile(Directory directory, const std::string& filePath, const std::string& data);
+
+        //Binary file read/write
+        std::vector<char> ReadBinaryFile(Directory directory, const std::string& filePath);
+        bool WriteBinaryFile(Directory directory, const std::string& filePath, const std::vector<char>& data);
+
+        std::string GetFilePath(Directory directory, const std::string& filePath);
+        bool FileExists(Directory directory, const std::string& filePath);
+
+        uint64_t LastModified(Directory directory, const std::string& filePath);
```

Comment:
not relevant to this PR, but i suggest we never use `int` but always use the `(u)intx_t` version so we're always explicit in what we want.

Reply:
I agree, I think they are more descriptive, and give away the type on first glance.